src/parseCss.ts

```diff
@@ -51,18 +51,37 @@ export const parseCss = (stylesContent: string, offset: number) => {
     try {
         usedRules = findUsedRules(stylesContent, offset)
     } catch (error) {
-        if (error.reason === 'Unknown word') {
-            const stylesWithoutErrorString = stylesContent
-                .split(/\n\r?/)
-                .filter((str, i) => i !== error.input.line - 1)
-                .join('\n')
-            try {
-                usedRules = findUsedRules(stylesWithoutErrorString, offset - error.input.endColumn)
-            } catch {}
+        if (error.name === 'CssSyntaxError' && error instanceof CssSyntaxError && error.reason === 'Unknown word') {
+            const { normalizedStyles, normalizedOffset } = normalizeStylesContent(stylesContent, offset, error)
+            usedRules = findUsedRules(normalizedStyles, normalizedOffset)
         }
     }
 
     return {
         usedRules,
     }
 }
+
+const normalizeStylesContent = (stylesContent: string, offset: number, error: CssSyntaxError) => {
+    const normalizedStyles = stylesContent
+        .split(/\n\r?/)
+        .filter((str, i) => {
+            if (error.input) {
+                return i !== error.input.line - 1
+            }
+
+            return true
+        })
+        .join('\n')
+    const normalizedOffset = offset - (error.input!.endColumn ?? 0)
+    try {
+        postcssParser(normalizedStyles)
+        return { normalizedStyles, normalizedOffset }
+    } catch (error) {
+        if (error.name === 'CssSyntaxError' && error instanceof CssSyntaxError && (error.reason === 'Unknown word' || error.reason.includes('Unexpected'))) {
+            return normalizeStylesContent(normalizedStyles, normalizedOffset, error)
```

Comment:
hm. just curios in what cases we need to do this recursevely? can't it go to ifinite call stack?

Reply:
Sorry missed comment above, probably we can leave it here as is but with safety check like maxAttemps, because since its running syncronously without a way to abort it, running it a lot of times can seriously slowdown extension host (especially on big styleheets).

Reply:
Ok, let's do it but it probably won't be able to handle it in cases like this:
```js
const Button = styled.button`
  padding-left: 1px;
  padding-right: 1px;
  display: flex;
  ${props =>
    props.primary
    }
  ${props =>
    props.primary &&
    css`
    padding-left: 1px;
    animation: inherit;
    padding-right: 1px;
  `}
  flex
`
```
I don't care actually

Reply:
btw 10 attempts are required to handle this case